modxcms/revolution
# Modernize manager template build workflow

### What does it do?

- Replaces uglifyjs with terser to support the current javascript standard (ECMA2015/ES6) and support future stable standards
- Bring other dev dependencies up to date
- Add back in the copyright banner for compiled JS and CSS files (the current config uses an option that was removed from cssmin/clean-css many years ago)
- Added new grunt task to just run the compile/minification routines for SASS/CSS
- Removed imageoptim, as the only image used in the interface of any consequence (size-wise) is the login background
- Made change to format of comments in the variables SASS file so they can be discarded during the build process
- Removed reference to source map file in the popper.min.js file to remove the console error it was causing (due to the map file not existing)

### Why is it needed?

Two reasons:
1. Virtually every tutorial and page of official documentation you will find on the web today prefers ES6 methods and support is robust among modern and semi-modern browsers. It's safe and prudent for MODX to adopt this standard, as its many great features allow more efficient coding and better scope control.
2. Dart SASS is the new and supported standard and the move to it will provide support for features that Node SASS will not (particularly moving into the future).

### How to test

1. Run _npm update_ in the _build/templates/default directory and run the _grunt build_, _style_, and _compress_ tasks to verify. 
2. Add a test method to the manager/assets/modext/util/utilities.js utilizing some ES6 syntax to verify it compiles correctly (running the compress or build task).
3. Verify in various target browsers that the changes do not adversely affect the manager's operation and styling looks the same as before the change.

Note that a warning will appear when compiling the SASS files, as fontawesome and normalize-scss use the to-be deprecated "/" for division. This will be replaced my the math.div([numerator], [denominator]) method and be required for Dart Sass 2.0+ (which is a ways off, as its current version is 1.37.5). This issue is on the radar for both repos.

### Related issue(s)/PR(s)
None
